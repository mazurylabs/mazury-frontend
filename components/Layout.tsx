import { useRouter } from 'next/router';
import React from 'react';
import Link from 'next/link';

export const Layout: React.FC = ({ children }) => {
  const router = useRouter();
  const { pathname } = router;
  const showSidebar = pathname !== '/';

  return (
    <div className='w-full flex'>
      {/* SIDEBAR ON THE LEFT */}
      {showSidebar && (
        <aside className='w-1/12 sticky top-0 border-2 h-screen py-10 px-5'>
          <menu className='flex flex-col gap-12 mt-12 text-xl font-bold items-center'>
            <Link href='/' passHref>
              <a>Home</a>
            </Link>
            <Link href='/search' passHref>
              <a>Search</a>
            </Link>
            <Link href='/people/0x' passHref>
              <a>Profile</a>
            </Link>
            <Link href='/settings' passHref>
              <a>Settings</a>
            </Link>
          </menu>
        </aside>
      )}

      {/* APP */}
      <main className='container w-11/12'>{children}</main>
    </div>
  );
};
