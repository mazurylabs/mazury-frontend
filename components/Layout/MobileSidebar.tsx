/* eslint-disable @next/next/no-img-element */

import Link from 'next/link';
import { useRouter } from 'next/router';
import * as React from 'react';
import clsx from 'clsx';
import SVG from 'react-inlinesvg';
import * as Popover from '@radix-ui/react-popover';

import { useLogout, useUser } from 'providers/react-query-auth';
import storage from 'utils/storage';
import { STORED_USER } from 'config';
import { useMobile, useUserSession } from 'hooks';
import { Avatar } from '../Avatar';

interface MobileSidebarProps {
  children?: React.ReactNode;
  className?: string;
}

export const MobileSidebar: React.FC<MobileSidebarProps> = ({
  children,
  className,
}) => {
  const userSessionExpired = useUserSession();
  const router = useRouter();
  const logout = useLogout();

  const storedUser = storage.getToken(STORED_USER);

  const { data: profile } = useUser({
    enabled: !userSessionExpired,
    initialData: storedUser,
  });

  return (
    <div
      className={clsx(
        'sticky bottom-0 left-0 flex w-screen items-center justify-center border bg-white px-[58.5px] py-4 lg:hidden',
        className
      )}
    >
      <div className="flex items-center space-x-16">
        <NavItem
          to="/"
          icon="/icons/opportunities.svg"
          label="home"
          active={
            router.pathname.includes('opportunities') || router.pathname === '/'
          }
        />
        <NavItem
          label="Search"
          to="/search"
          icon="/icons/search.svg"
          active={router.pathname.includes('search')}
        />

        {storedUser?.is_recruiter && (
          <NavItem
            label="Projects"
            to="/projects"
            icon="/icons/dashboard.svg"
            active={router.pathname.includes('projects')}
          />
        )}

        {!storedUser && (
          <Link href="/sign-in" passHref>
            <span className="sr-only">Sign in</span>
            <SVG
              src="/icons/user.svg"
              width={24}
              height={24}
              className={
                router.pathname.includes('sign-in')
                  ? 'text-indigoGray-90'
                  : 'text-indigoGray-40'
              }
            />
          </Link>
        )}

        {storedUser && (
          <Popover.Root>
            <div className="flex items-center space-x-4">
              <Popover.Trigger className="relative">
                <Avatar
                  src={profile?.avatar || '/profile-active.svg'}
                  alt={profile?.username || ''}
                  variant="sm"
                />
              </Popover.Trigger>
            </div>

            <Popover.Portal>
              <Popover.Content
                align="center"
                sideOffset={10}
                className="h-[106px] w-[200px]"
              >
                <div className="flex flex-col py-[5px] space-y-[6.5px] items-start h-full w-full flex-col rounded-lg bg-white shadow-base font-sans text-sm text-indigoGray-90">
                  <Link
                    href={`/people/${storedUser.eth_address}`}
                    passHref
                    className="font-regular pl-4 grow text-sm font-sans text-indigoGray-90 flex items-center space-x-3 w-full grow"
                  >
                    <Avatar
                      src={storedUser.avatar}
                      alt="user"
                      variant="xs"
                      outerClassName="h-4 w-4 shrink-0"
                    />
                    <span>Profile</span>
                  </Link>

                  <Link
                    href={'/settings'}
                    passHref
                    className="font-regular pl-4 grow text-sm font-sans text-indigoGray-90 flex items-center space-x-3 w-full grow"
                  >
                    <SVG src="/icons/settings.svg" width={16} height={16} />
                    <span>Settings</span>
                  </Link>

                  <button
                    type="button"
                    className="font-regular pl-4 grow text-sm font-sans text-indigoGray-90 flex items-center space-x-3 w-full grow"
                    onClick={() =>
                      window.open('https://mazury.canny.io/feedback', '_blank')
                    }
                  >
                    <SVG src="/icons/help-circle.svg" width={16} height={16} />
                    <span>Give feedback</span>
                  </button>

                  <button
                    type="button"
                    className="font-regular pl-4 grow text-sm font-sans text-indigoGray-90 flex items-center space-x-3 w-full grow"
                    onClick={() =>
                      logout.mutate({}, { onSuccess: () => router.push('/') })
                    }
                  >
                    <SVG src="/icons/sign-out.svg" width={16} height={16} />
                    <span>Sign out</span>
                  </button>
                </div>
                <Popover.Arrow fill="white" className="drop-shadow" />
              </Popover.Content>
            </Popover.Portal>
          </Popover.Root>
        )}
      </div>
    </div>
  );
};

const NavItem = ({
  to,
  active,
  icon,
  label,
}: {
  to: string;
  active: boolean;
  icon: string;
  label: string;
}) => {
  return (
    <Link
      href={to}
      passHref
      className={clsx(active ? 'text-indigoGray-90' : 'text-indigoGray-40')}
    >
      <span className="sr-only">{label}</span>
      <SVG src={icon} width={24} height={24} />
    </Link>
  );
};
