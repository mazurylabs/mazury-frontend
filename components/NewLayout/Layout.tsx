import { motion } from 'framer-motion';
import { FC, useState } from 'react';
import { SidebarContext } from '../../contexts';

interface LayoutProps {
  sidebarContent: React.ReactNode;
  innerLeftContent: React.ReactNode;
  innerRightContent: React.ReactNode;
  headerContent?: React.ReactNode;
}

export const Layout: FC<LayoutProps> = ({
  sidebarContent,
  innerLeftContent,
  innerRightContent,
  headerContent,
}) => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <div className='flex w-full min-h-screen'>
      <SidebarContext.Provider value={{ isOpen, setIsOpen }}>
        <motion.aside
          className={`fixed w-[75px] top-0 border h-screen py-10 px-5 shadow-inner flex flex-col z-20`}
          style={{
            background:
              'linear-gradient(184.64deg, #FFFFFF -5.13%, #E7E8E9 103.54%)',
          }}
          whileHover={{
            width: '266px',
          }}
          onMouseEnter={() => setIsOpen(true)}
          onMouseLeave={() => setIsOpen(false)}
        >
          {sidebarContent}
        </motion.aside>
      </SidebarContext.Provider>

      <main className='ml-[75px] flex flex-col w-11/12 min-h-screen border gap-8 p-8'>
        {headerContent && <div className='border'>headerContent</div>}

        <div className='flex gap-8'>
          <div className='flex flex-col w-3/12 border gap-8'>
            {innerLeftContent}
          </div>

          <div className='flex flex-col w-9/12 border gap-8'>
            {innerRightContent}
          </div>
        </div>
      </main>
    </div>
  );
};
