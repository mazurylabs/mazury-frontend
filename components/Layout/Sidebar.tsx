import Link from 'next/link';
import * as React from 'react';
import { useRouter } from 'next/router';
import clsx from 'clsx';
import SVG from 'react-inlinesvg';

import { verifyEmail } from 'utils/api';
import { Avatar, WalletRequestModal } from 'components';
import { SignIn } from 'views/SignIn';
import { useLogout, useUser } from 'providers/react-query-auth';
import { useUserSession } from 'hooks';
import storage from 'utils/storage';
import { STORED_USER } from 'config';
import { SidebarContext } from 'contexts';

type Steps = 'idle' | 'active' | 'error';

export const Sidebar: React.FC = () => {
  const { isOpen, signInOpen, setIsOpen, setSignInOpen } =
    React.useContext(SidebarContext);

  const logout = useLogout();
  const router = useRouter();

  const userSessionExpired = useUserSession();
  const [currentStep, setCurrentStep] = React.useState<Steps>('idle');

  const storedUser = storage.getToken(STORED_USER);

  const { data: profile } = useUser({
    enabled: !userSessionExpired,
    initialData: storedUser,
  });

  const navItems = {
    secondRow: [
      {
        name: 'Opportunities',
        to: '/',
        icon: '/icons/opportunities.svg',
        active:
          router.pathname.includes('opportunities') || router.pathname === '/',
      },
      {
        name: 'Search',
        to: '/search',
        icon: '/icons/search.svg',
        active: router.pathname.includes('search'),
      },
      storedUser?.is_recruiter && {
        name: 'Projects',
        to: '/projects',
        icon: '/icons/dashboard.svg',
        active: router.pathname.includes('projects'),
      },
    ].filter(Boolean),
    thirdRow: [
      storedUser && {
        name: 'Profile',
        to: `/people/${storedUser.eth_address}`,
        icon: (
          <Avatar
            src={storedUser.avatar}
            alt="user"
            variant="xs"
            outerClassName="h-4 w-4 shrink-0"
          />
        ),
        active: router.pathname.includes('people'),
      },
      !storedUser && {
        name: 'Sign in',
        onClick: () => setSignInOpen(true),
        icon: '/icons/user.svg',
      },
      storedUser && {
        name: 'Settings',
        to: `/settings`,
        icon: '/icons/settings.svg',
        active: router.pathname.includes('settings'),
      },
      {
        name: 'Give feedback',
        icon: '/icons/help-circle.svg',
        active: false,
        onClick: () =>
          window.open('https://mazury.canny.io/feedback', '_blank'),
      },
      storedUser && {
        name: 'Sign out',
        onClick: () => logout.mutate({}, { onSuccess: () => router.push('/') }),
        icon: '/icons/sign-out.svg',
      },
    ].filter(Boolean),
  };

  const handleEmailVerification = async () => {
    if (profile?.eth_address) {
      const { error } = await verifyEmail(profile?.eth_address);
      if (!error) {
        setCurrentStep('idle');
      } else {
        setCurrentStep('error');
      }
    }
  };

  return (
    <>
      <aside
        role="menu"
        className={clsx(
          'h-screen fixed left-0 top-0 z-30 hidden lg:flex bg-white px-2 pt-6 pb-4 border-r border-indigoGray-20',
          signInOpen && 'w-[335px]'
        )}
        onPointerEnter={() => setIsOpen(true)}
        onPointerLeave={() => !signInOpen && setIsOpen(false)}
      >
        {signInOpen && (
          <div className="flex flex-col">
            <button
              type="button"
              className="justify-start hover:cursor-pointer h-6 w-6"
              aria-label="back"
              onClick={() => setSignInOpen(false)}
            >
              <SVG src="/icons/arrow-left.svg" height={24} width={24} />
            </button>

            <SignIn />
          </div>
        )}

        {!signInOpen && (
          <div
            className={clsx(
              'flex flex-col grow transition-all',
              isOpen ? 'w-[184px]' : 'w-10'
            )}
          >
            <div className="flex w-full items-center">
              <SidebarItem
                icon={
                  <div className="h-8 w-8">
                    <SVG src="/new-logo.svg" className="h-8 w-8 shrink-0" />
                  </div>
                }
                href="/"
              />
            </div>

            <div className="grow border-t border-b border-indigoGray-20 pt-8 my-8 space-y-4">
              {navItems.secondRow.map((item) => (
                <SidebarItem
                  key={item.name}
                  icon={item.icon}
                  label={item.name}
                  href={item.to}
                  active={item.active}
                />
              ))}
            </div>

            <div className="space-y-3">
              {navItems.thirdRow.map((item) => (
                <SidebarItem
                  key={item.name}
                  icon={item.icon}
                  label={item.name}
                  href={item.to}
                  onClick={item.onClick}
                  active={item.active}
                  variant="sm"
                />
              ))}
            </div>
          </div>
        )}
      </aside>

      {signInOpen && (
        <WalletRequestModal
          step={currentStep}
          handleSkip={() => setCurrentStep('idle')}
          handleRequestSignature={handleEmailVerification}
        />
      )}
    </>
  );
};

type SidebarItemProps = {
  label?: string;
  icon: React.ReactNode | string;
  active?: boolean;
  href: string;
  onClick?: () => void;
  variant?: 'sm' | 'lg';
};

const SidebarItem = ({
  label,
  icon,
  active = false,
  variant = 'lg',
  ...props
}: SidebarItemProps) => {
  const Component = props.href ? Link : 'button';

  return (
    <Component
      {...props}
      className={clsx(
        'transition-all rounded-md flex items-center space-x-3 font-sans text-sm overflow-hidden w-full',
        variant === 'lg' ? 'h-10 font-medium' : 'h-[16px] font-normal',
        label && 'px-3',
        label &&
          !active &&
          variant === 'lg' &&
          'hover:bg-indigoGray-10 hover:text-indigoGray-50',
        label && !active && variant === 'sm' && 'hover:text-indigoGray-50',
        active && variant === 'lg' && 'bg-indigoGray-90 text-indigo-50',
        active && variant === 'sm' && 'text-indigoGray-50',
        !active && variant === 'lg' && 'bg-transparent text-indigoGray-90'
      )}
    >
      {typeof icon === 'string' ? (
        <SVG src={icon} className="shrink-0 h-4 w-4" />
      ) : (
        icon
      )}{' '}
      <span className="whitespace-nowrap">{label}</span>
    </Component>
  );
};
