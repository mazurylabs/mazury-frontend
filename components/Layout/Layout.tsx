import { motion } from 'framer-motion';
import { FC, useContext } from 'react';
import { SidebarContext } from 'contexts';
import { MobileSidebar } from '../MobileSidebar/MobileSidebar';
import { Sidebar } from 'components';
import Link from 'next/link';
import SVG from 'react-inlinesvg';
import clsx from 'clsx';

interface LayoutProps {
  sidebarContent?: React.ReactNode;
  innerLeftContent?: React.ReactNode;
  innerRightContent?: React.ReactNode;
  headerContent?: React.ReactNode;
  variant?: 'three-part' | 'plain';
  children?: React.ReactNode;
  showMobileSidebar?: boolean;
  className?: string;
}

export const Layout: FC<LayoutProps> = ({
  sidebarContent = <Sidebar />,
  innerLeftContent,
  innerRightContent,
  headerContent,
  variant = 'three-part',
  children,
  showMobileSidebar = true,
  className,
}) => {
  const { isOpen, signInOpen, setIsOpen } = useContext(SidebarContext);

  const variants = {
    open: {
      width: 225,
      transition: { duration: 0.2 },
    },
    signInOpen: {
      width: 335,
      transition: { duration: 0.2 },
    },
    closed: {
      width: 75,
      transition: { duration: 0.2 },
    },
  };

  return (
    <div
      className="flex min-h-screen w-full flex-col"
      data-testid="layout-container"
    >
      <motion.aside
        variants={variants}
        animate={signInOpen ? 'signInOpen' : isOpen ? 'open' : 'closed'}
        className={`fixed left-0 top-0 z-30 !hidden h-screen w-[75px] flex-col items-center bg-white px-4 py-6 shadow-inner lg:!flex`}
        onMouseEnter={() => setIsOpen(true)}
        onMouseLeave={() => !signInOpen && setIsOpen(false)}
        role="menu"
      >
        <Link legacyBehavior href="/">
          <a className="h-[32px] w-[32px] cursor-pointer">
            <SVG src="/new-logo.svg" height="32px" width="32px" />
          </a>
        </Link>

        <hr className={`mx-3 my-8 w-full border border-indigoGray-20`} />
        {sidebarContent}
      </motion.aside>

      <main
        className={clsx(
          'mx-auto flex  w-full grow flex-col gap-8 px-0 pt-0 md:px-8 lg:ml-[75px] lg:w-[calc(100vw-75px)]',
          className
        )}
      >
        {variant === 'three-part' && (
          <>
            {headerContent}

            <div className="flex min-h-[80vh] grow flex-col gap-8 md:flex-row">
              {/* TODO: Fix alignment. Align with the content in the header */}
              <div className="hidden w-full flex-col gap-4 md:flex md:w-3/12">
                {innerLeftContent}
              </div>

              <div className="flex w-full grow flex-col gap-8 px-4 md:w-9/12 md:px-0">
                {innerRightContent}
              </div>
            </div>
          </>
        )}
        {variant === 'plain' && <>{children}</>}
      </main>

      <MobileSidebar className={clsx(!showMobileSidebar && 'hidden')} />
    </div>
  );
};
