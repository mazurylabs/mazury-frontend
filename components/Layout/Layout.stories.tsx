import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';
import { Sidebar } from '../Sidebar';

import { Layout } from './Layout';

export default {
  title: 'Components/Layout',
  component: Layout,
  argTypes: {
    sidebarContent: {
      control: { type: 'text' },
    },
    innerLeftContent: {
      control: { type: 'text' },
    },
    innerRightContent: {
      control: { type: 'text' },
    },
    headerContent: {
      control: { type: 'text' },
    },
  },
} as ComponentMeta<typeof Layout>;

const Template: ComponentStory<typeof Layout> = (args) => <Layout {...args} />;

export const Default = Template.bind({});
Default.args = {
  sidebarContent: <Sidebar />,
  innerLeftContent: <h1>Inner left</h1>,
  innerRightContent: <h1>Inner right</h1>,
};

export const WithHeader = Template.bind({});
WithHeader.args = {
  sidebarContent: <Sidebar />,
  innerLeftContent: <h1>Inner left</h1>,
  innerRightContent: <h1>Inner right</h1>,
  headerContent: <h1>header</h1>,
};
