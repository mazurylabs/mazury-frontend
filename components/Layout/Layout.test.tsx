import { render, screen } from '@testing-library/react';
// TODO: Mock the router and test with the actual sidebar
// import { Sidebar } from 'components';
import { Layout } from './Layout';

describe('Layout', () => {
  test('all parts are being rendered correctly', () => {
    render(
      <Layout
        sidebarContent={<Sidebar />}
        innerLeftContent={<span>Inner left content</span>}
        innerRightContent={<span>Inner right content</span>}
        headerContent={<span>Header content</span>}
      />
    );
    const sidebar = screen.getByRole('menu');
    expect(sidebar).toBeInTheDocument();
    const innerLeft = screen.getByText('Inner left content');
    expect(innerLeft).toBeInTheDocument();
    const innerRight = screen.getByText('Inner right content');
    expect(innerRight).toBeInTheDocument();
    const header = screen.getByText('Header content');
    expect(header).toBeInTheDocument();
    const layoutContainer = screen.getByTestId('layout-container');
    expect(layoutContainer).toBeInTheDocument();
    expect(layoutContainer).toMatchSnapshot();
  });
});
