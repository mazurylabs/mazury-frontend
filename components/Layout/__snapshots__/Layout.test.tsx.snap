// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout all parts are being rendered correctly 1`] = `
<div
  class="flex min-h-screen w-full"
  data-testid="layout-container"
>
  <aside
    class="fixed top-0 z-20 hidden h-screen w-[75px] flex-col bg-white py-10 false shadow-inner lg:flex"
    role="menu"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzIiIGhlaWdodD0iMzIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Mazury logo"
        class="mx-auto rounded-full hover:cursor-pointer"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <hr
      class="border border-indigoGray-20 mx-3 mt-8"
    />
    <menu
      class="mt-12 flex flex-col text-xl font-bold items-center h-[100%] px-3"
    >
      <a
        class="flex h-[40px] false items-center gap-4 rounded-md border border-hidden p-3 text-sm font-medium text-indigoGray-90 hover:bg-indigoGray-10 hover:text-indigoGray-50 active:border-solid active:border-indigoGray-30 active:bg-indigoGray-10 active:text-indigoGray-80 false undefined"
        href="/"
      >
        <svg
          aria-label="Search icon"
          fill="none"
          height="16px"
          role="img"
          viewBox="0 0 16 16"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.33333 12.6667C10.2789 12.6667 12.6667 10.2789 12.6667 7.33333C12.6667 4.38781 10.2789 2 7.33333 2C4.38781 2 2 4.38781 2 7.33333C2 10.2789 4.38781 12.6667 7.33333 12.6667Z"
            stroke="#110F2A"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.33"
          />
          <path
            d="M14 14.0001L11.1 11.1001"
            stroke="#110F2A"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.33"
          />
        </svg>
         
      </a>
      <a
        class="flex h-[40px] false items-center gap-4 rounded-md border false p-3 text-sm font-medium text-indigoGray-90 hover:bg-indigoGray-10 hover:text-indigoGray-50 active:border-solid active:border-indigoGray-30 active:bg-indigoGray-10 active:text-indigoGray-80 border-solid border-indigoGray-30 bg-indigoGray-90 text-indigo-50 mt-4"
        href="/"
      >
        <svg
          aria-label="Home icon"
          fill="none"
          height="16px"
          role="img"
          viewBox="0 0 16 16"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 6L8 1.33333L14 6V13.3333C14 13.687 13.8595 14.0261 13.6095 14.2761C13.3594 14.5262 13.0203 14.6667 12.6667 14.6667H3.33333C2.97971 14.6667 2.64057 14.5262 2.39052 14.2761C2.14048 14.0261 2 13.687 2 13.3333V6Z"
            stroke="#eef2ff"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.33"
          />
        </svg>
         
      </a>
      <div
        class="mt-auto flex flex-col"
      >
        <hr
          class="border border-indigoGray-20 undefined"
        />
        <a
          class="flex h-[40px] false items-center gap-4 rounded-md border border-hidden p-3 text-sm font-medium text-indigoGray-90 hover:bg-indigoGray-10 hover:text-indigoGray-50 active:border-solid active:border-indigoGray-30 active:bg-indigoGray-10 active:text-indigoGray-80 false mt-8"
          href="/people/0x123"
        >
          <img
            alt="Profile icon"
            src="/profile-active.svg"
          />
           
        </a>
        <a
          class="flex h-[40px] false items-center gap-4 rounded-md border border-hidden p-3 text-sm font-medium text-indigoGray-90 hover:bg-indigoGray-10 hover:text-indigoGray-50 active:border-solid active:border-indigoGray-30 active:bg-indigoGray-10 active:text-indigoGray-80 false mb-4 mt-4"
          href="/settings"
        >
          <svg
            aria-label="Settings icon"
            fill="none"
            height="16px"
            role="img"
            viewBox="0 0 16 16"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.66667 14V9.33334"
              stroke="#110F2A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
            <path
              d="M2.66667 6.66667V2"
              stroke="#110F2A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
            <path
              d="M8 14V8"
              stroke="#110F2A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
            <path
              d="M8 5.33333V2"
              stroke="#110F2A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
            <path
              d="M13.3333 14V10.6667"
              stroke="#110F2A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
            <path
              d="M13.3333 8V2"
              stroke="#110F2A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
            <path
              d="M0.666668 9.33334H4.66667"
              stroke="#110F2A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
            <path
              d="M6 5.33334H10"
              stroke="#110F2A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
            <path
              d="M11.3333 10.6667H15.3333"
              stroke="#110F2A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
          </svg>
           
        </a>
      </div>
    </menu>
  </aside>
  <main
    class="mx-auto flex min-h-[90vh] w-full flex-col gap-8 px-0 pt-0 md:px-8 lg:ml-[75px] lg:w-11/12"
  >
    <span>
      Header content
    </span>
    <div
      class="flex grow flex-col gap-8 md:flex-row"
    >
      <div
        class="hidden w-full flex-col gap-4 md:flex md:w-3/12"
      >
        <span>
          Inner left content
        </span>
      </div>
      <div
        class="flex w-full grow flex-col gap-8 px-4 md:w-9/12 md:px-0"
      >
        <span>
          Inner right content
        </span>
      </div>
    </div>
  </main>
  <div
    class="sticky bottom-0 left-0 flex w-screen items-center justify-between border bg-white px-[58.5px] pt-4 pb-8 lg:hidden"
  >
    <a
      href="/"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Home icon"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </a>
    <a
      href="/search"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Search icon"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </a>
    <a
      href="/people/0x123"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Profile icon"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
       
    </a>
    <a
      href="/settings"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Settings icon"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
       
    </a>
  </div>
</div>
`;
